chisel/src/main.rs

```diff
@@ -8,400 +8,110 @@ extern crate serde_yaml;
-    if let Ok(buffer) = read(context.file()) {
-        if let Ok(module) = deserialize_buffer::<Module>(&buffer) {
-            // If we do not parse the NamesSection here, parity-wasm will drop it at serialisation
-            // It is useful to have this for a number of optimisation passes, including binaryenopt and snip
-            // TODO: better error handling
-            let mut module = module.parse_names().expect("Failed to parse NamesSection");
-
-            let original = module.clone();
-            eprintln!("Ruleset {}:", context.name());
-            let chisel_results = context
-                .get_modules()
-                .iter()
-                .map(|ctx| execute_module(ctx, &mut module))
-                .fold(true, |b, e| e & b);
-
-            // If the module was mutated, serialize to file.
-            if original != module {
-                if let Some(path) = context.outfile() {
-                    chisel_debug!(1, "Writing to file: {}", path);
-                    serialize_to_file(path, module).unwrap();
-                } else {
-                    chisel_debug!(1, "No output file specified; writing in place");
-                    serialize_to_file(context.file(), module).unwrap();
-                }
-            }
-            Ok(chisel_results)
-        } else {
-            Err(ERR_DESERIALIZE_MODULE)
-        }
-    } else {
-        Err(ERR_FAILED_OPEN_BINARY)
-    }
-}
-
-fn chisel_subcommand_run(args: &ArgMatches) -> i32 {
-    let config_path = args.value_of("CONFIG").unwrap_or(DEFAULT_CONFIG_PATH);
-
-    if args.is_present("VERBOSE") {
-        logger::set_global_log_level(1);
-    }
-
-    if let Ok(conf) = read_to_string(config_path) {
-        let ctxs = yaml_configure(&conf);
-        chisel_debug!(1, "Loaded {} rulesets", ctxs.len());
-
-        let result_final = ctxs.iter().fold(0, |acc, ctx| {
-            chisel_debug!(1, "Executing ruleset {}", ctx.name());
-            match chisel_execute(&ctx) {
-                Ok(result) => acc + if result { 0 } else { 1 }, // Add the number of module failures to exit code.
-                Err(msg) => err_exit(msg),
-            }
-        });
-
-        result_final
-    } else {
-        err_exit(&format!(
-            "Could not load configuration file: {}",
-            config_path
-        ))
-    }
-}
-
-pub fn main() {
+fn main() {
     let cli_matches = App::new("chisel")
         .version(crate_version!())
         .about(crate_description!())
         .arg(
-            Arg::with_name("VERBOSE")
-                .short("v")
-                .long("verbose")
-                .help("Enables verbose debug logging")
+            Arg::with_name("NO_RECOVER")
+                .short("x")
+                .long("norecover")
+                .help("Exits immediately on all recoverable errors")
                 .global(true),
         )
+        .arg(
+            Arg::with_name("DEBUG_MESSAGES")
```

Comment:
Why not keep `VERBOSE` as before?

Reply:
Looks strange when you run `chisel --help`